src/OrchardCore.Modules/Orchard.OpenId/Services/OpenIdCorsConfiguration.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Linq;
+using Microsoft.AspNetCore.Cors.Infrastructure;
+using Microsoft.AspNetCore.Modules;
+using Microsoft.Extensions.Options;
+
+namespace Orchard.OpenId.Services
+{
+    [RequireFeatures("Orchard.Cors.Mvc")]
+    public class OpenIdCorsConfiguration : IConfigureOptions<CorsOptions>
+    {
+        private readonly IOpenIdService _openIdService;
+
+        public OpenIdCorsConfiguration(IOpenIdService openIdService)
+        {
+            _openIdService = openIdService;
+        }
+
+        public void Configure(CorsOptions options)
+        {
+            var openIdSettings = _openIdService.GetOpenIdSettingsAsync().GetAwaiter().GetResult();
+            if (openIdSettings.Audiences != null && openIdSettings.Audiences.Any())
+                options.AddPolicy(Constants.OpenIdConnectPolicy, builder => builder
+                    .WithOrigins(openIdSettings.Audiences.ToArray())
+                    .AllowAnyHeader()
+                    .AllowAnyMethod()
+                    .AllowCredentials()
```

Comment:
Flows based on redirects don't require Cors for the `AccessController`, only for the `UserInfo` controller. We are also assuming that every configured App requires UserInfo. So, we'd have to enable/disable Cors at the App level (or for each audience)  and have 2 policies: one for the AccessController and another for the UserController. Also, not all flows require `AllowCredentials`.